### PR TITLE
fix(mocks): Use java.lang.Object if there are protos named 'Object' [ggj]

### PR DESCRIPTION
**Pull request by miraleung**

### CODE REVIEW

**Review by noahdietz — APPROVED** [review]

File: src/main/java/com/google/api/generator/gapic/composer/grpc/MockServiceImplClassComposer.java [hunk 1]
```diff
@@ -97,12 +90,34 @@ public static MockServiceImplClassComposer instance() {
...
     TypeStore typeStore = createDynamicTypes(service);
     String className = ClassNames.getMockServiceImplClassName(service);
     GapicClass.Kind kind = Kind.TEST;
     String pakkage = service.pakkage();
 
+    // Use the full name java.lang.Object if there is a proto message that is also named "Object".
+    // Affects GCS.
+    if (context.messages().keySet().stream()
+        .map(s -> s.substring(Math.max(0, s.lastIndexOf(".") + 1), s.length()))
+        .collect(Collectors.toSet())
```
Comment (vam-google) [inline src/main/java/com/google/api/generator/gapic/composer/grpc/MockServiceImplClassComposer.java:103]: Looks like you don't really need the collected set of stuff, you just curious if three is any object of name "Object", so please consider something shorter, like:
```java
if(context.messages().keySet().stream().anyMatch(s -> s.endsWith(".Object"))) {}
```

**Review by vam-google — COMMENTED** [review]
LGTM with few minor comments

File: src/main/java/com/google/api/generator/gapic/composer/grpc/MockServiceImplClassComposer.java [hunk 2]
```diff
@@ -97,12 +90,34 @@ public static MockServiceImplClassComposer instance() {
...
+  public GapicClass generate(GapicContext context, Service service) {
     TypeStore typeStore = createDynamicTypes(service);
     String className = ClassNames.getMockServiceImplClassName(service);
     GapicClass.Kind kind = Kind.TEST;
     String pakkage = service.pakkage();
 
+    // Use the full name java.lang.Object if there is a proto message that is also named "Object".
+    // Affects GCS.
+    if (context.messages().keySet().stream()
+        .map(s -> s.substring(Math.max(0, s.lastIndexOf(".") + 1), s.length()))
```
[the remark this answers is not in the document]
Comment (miraleung) [reply src/main/java/com/google/api/generator/gapic/composer/grpc/MockServiceImplClassComposer.java]: Done.

File: src/main/java/com/google/api/generator/gapic/composer/grpc/MockServiceImplClassComposer.java (hunk 1 above)
Comment (miraleung) [reply src/main/java/com/google/api/generator/gapic/composer/grpc/MockServiceImplClassComposer.java]: Done.

**Pull request merged**